src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AdoNet/SqlClient/SqlCommandExecuteScalarAsyncIntegration.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Data.Common;
+using System.Threading;
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet.SqlClient
+{
+    /// <summary>
+    /// CallTarget instrumentation for:
+    /// Task[object] System.Data.SqlClient.SqlCommand.ExecuteScalarAsync(CancellationToken)
+    /// Task[object] Microsoft.Data.SqlClient.SqlCommand.ExecuteScalarAsync(CancellationToken)
+    /// </summary>
+    [InstrumentMethod(
+        Assemblies = new[] { SqlClientConstants.SystemData.AssemblyName, SqlClientConstants.SystemData.SqlClientAssemblyName },
+        Type = SqlClientConstants.SystemData.SqlCommandType,
+        Method = AdoNetConstants.MethodNames.ExecuteScalarAsync,
+        ReturnTypeName = "System.Threading.Tasks.Task`1<System.Object>",
+        ParametersTypesNames = new[] { ClrNames.CancellationToken },
+        MinimumVersion = SqlClientConstants.SystemData.MinimumVersion,
+        MaximumVersion = SqlClientConstants.SystemData.MaximumVersion,
+        IntegrationName = SqlClientConstants.SqlCommandIntegrationName)]
+    [InstrumentMethod(
+        Assembly = SqlClientConstants.MicrosoftData.AssemblyName,
+        Type = SqlClientConstants.MicrosoftData.SqlCommandType,
+        Method = AdoNetConstants.MethodNames.ExecuteScalarAsync,
+        ReturnTypeName = "System.Threading.Tasks.Task`1<System.Object>",
+        ParametersTypesNames = new[] { ClrNames.CancellationToken },
+        MinimumVersion = SqlClientConstants.MicrosoftData.MinimumVersion,
+        MaximumVersion = SqlClientConstants.MicrosoftData.MaximumVersion,
+        IntegrationName = SqlClientConstants.SqlCommandIntegrationName)]
+    public class SqlCommandExecuteScalarAsyncIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="cancellationToken">CancellationToken value</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget>(TTarget instance, CancellationToken cancellationToken)
```

Comment:
I love how small the integration code is now, aside from starting/decorating a new Scope 😍

Reply:
Indeed. These files are mostly huge `[InstrumentMethod]` attributes now, which makes think we should look into improving this somehow.

Reply:
See my latest commit 😉